torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -386,6 +396,314 @@ at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
       at::device(kCUDA).dtype(dtype)
   );
 }
+
+
+template <typename T>
+std::string get_vector_str(std::vector<T> & vec, int64_t start, int64_t end=-1) {
+  /*
+  A utility function for returning a string representation of a std::vector containing ints/floats.
+  Useful for printing the shape of an N-dimensional tensor. This is used in the error message creation
+  part in the _extract_ndarrays function below.
+  */
+  std::ostringstream output;
+  output << "(";
+
+  if (end < 0) {
+    end = vec.size();
+  }
+  for (int64_t i = start; i < end; ++i) {
+    if (i != end - 1) {
+      output << vec[i] << ", ";        
+    } else {
+      output << vec[i] << ")";
+    }
+  }
+  return output.str();
+}  
+
+template <typename T>
+std::string get_buffer_str(void* buffer, int64_t start, int64_t end) {
```

Comment:
Let's just add one of these two printing functions. This is probably the one to keep. The vector version can call this by passing its data ptr, start and end.

Reply:
I have made ```get_vector_str``` use ```get_buffer_str``` for creating the string representation.